### PR TITLE
LPS-68398 Users are unable to navigate to sites that they are members of through their user group membership

### PR DESCRIPTION
**Pull request by pei-jung**

### CODE REVIEW

**Comment by JorgeFerrer** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: portal-impl/test/integration/com/liferay/portlet/service/persistence/GroupFinderTest.java [hunk 1]
```diff
@@ -170,6 +181,44 @@ public void testFindByCompanyId() throws Exception {
 	}
 
 	@Test
+	public void testFindByCompanyIdByUserGroupGroup() throws Exception {
+		_userGroup = UserGroupTestUtil.addUserGroup();
+		_userGroupGroup = GroupTestUtil.addGroup();
+		_userGroupUser = UserTestUtil.addUser();
+
+		GroupLocalServiceUtil.addUserGroupGroup(
```
Comment (JorgeFerrer) [inline portal-impl/test/integration/com/liferay/portlet/service/persistence/GroupFinderTest.java]: Hey @pei-jung,

Is this really necessary? When creating a UserGroup it automatically gets an associated Group. You should not have to manually create one and associate it.

Comment (pei-jung) [inline portal-impl/test/integration/com/liferay/portlet/service/persistence/GroupFinderTest.java]: Hey @JorgeFerrer,

We need to manually create a site and associate it to test site membership inheritance from user group. These site groups have classNameId of `Group.class`. The group that's created automatically when creating a UserGroup represents the UserGroup entity and has a different classNameId of `UserGroup.class`.  Currently, the code does not take the UserGroup group into account, so we can't use the associated group for the test.

**Comment by pei-jung** [comment]
@JorgeFerrer,

I've gone ahead and sent the pr to Brian https://github.com/brianchandotcom/liferay-portal/pull/43621.

File: portal-impl/test/integration/com/liferay/portlet/service/persistence/GroupFinderTest.java (hunk 1 above)
Comment (JorgeFerrer) [inline portal-impl/test/integration/com/liferay/portlet/service/persistence/GroupFinderTest.java]: Hey Pei-Jung,

I'm very confused by that answer. What you are doing in the test is not what is being done when the User Group is created from the UI, is it?

I'm not sure I understand why you need to change the classNameId. We can talk about it on Slack if you prefer.

**Pull request closed without merging**